SingularityUI/app/controllers/DeployDetail.es6

```diff
@@ -0,0 +1,45 @@
+import Controller from './Controller';
+import DeployView from '../views/deploy';
+import { FetchAction as DeployFetchAction} from '../actions/api/deploy';
+import { fetchTask as TaskFetchAction} from '../actions/api/task';
+import { FetchForDeployAction } from '../actions/api/tasks';
+import { FetchForDeploy as TaskHistoryFetchForDeploy } from '../actions/api/taskHistory';
+
+class DeployDetailController extends Controller {
+
+    initialize({store, requestId, deployId}) {
+        app.showPageLoader()
+        this.title('Status');
+        this.store = store;
+        this.requestId = requestId;
+        this.deployId = deployId;
+
+        let initPromise = this.store.dispatch(DeployFetchAction.trigger(requestId, deployId));
+        initPromise.then(() => {
+          this.setView(new DeployView(store));
+          app.showView(this.view);
+        });
+
+        let tasksPromise = this.store.dispatch(FetchForDeployAction.trigger(requestId, deployId));
+        tasksPromise.then(() => {
+          for (let t of store.getState().api.activeTasksForDeploy.data) {
+            this.store.dispatch(TaskFetchAction(t.taskId.id));
+          }
+        });
+
+        this.store.dispatch(TaskHistoryFetchForDeploy.trigger(requestId, deployId, 5, 1));
```

Comment:
it'd be a better experience for the user if we held of calling `showView()` until all the promises (or atleast, all the ones for visible things) have been resolved. i'd suggest coalescing the individual promises into an uber-promise using `Promise.all()`. and then calling `showView()` once the uber-promise has resolved.